edns_test.go

```diff
@@ -15,6 +15,18 @@ func TestOPTTtl(t *testing.T) {
 	if !e.Do() {
 		t.Errorf("DO bit should be non-zero")
 	}
+	e.SetDo(true)
+	if !e.Do() {
+		t.Errorf("DO bit should still be non-zero")
+	}
+	e.SetDo(false)
+	if e.Do() {
+		t.Errorf("DO bit should be zero")
+	}
+	e.SetDo(false)
```

Comment:
Actually this was tested, but there was a missing test or two. I made a comment explaining comprehensive tests for this, and annotated each call to explain which case is covered. Hopefully that makes it clear (and hopefully I didn't mess it up).